### ISSUE TITLE
[BUG] `ml-commons` and `ml-client` export the same symbols.

### ISSUE THREAD

**Issue by Yury-Fridlyand** [opensearch-project/ml-commons#778]
## What is the bug?
`ml-client` contains jar entire `ml-common`. This produces jar hell when I try to use both of them as dependencies.

## How can one reproduce the bug?
1. Build **non**-shadow jars for all projects to local maven repo
2. Build SQL plugin using local maven as a source for all `ml-everything`
3. Try to install it

```
Exception in thread "main" java.lang.IllegalStateException: failed to load plugin opensearch-sql due to jar hell
        at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:681)
        at org.opensearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:862)
        at org.opensearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:830)
        at org.opensearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:875)
        at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:275)
        at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:249)
        at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104)
        at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
        at org.opensearch.cli.MultiCommand.execute(MultiCommand.java:104)
        at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
        at org.opensearch.cli.Command.main(Command.java:101)
        at org.opensearch.plugins.PluginCli.main(PluginCli.java:66)
Caused by: java.lang.IllegalStateException: jar hell!
class: org.opensearch.ml.common.annotation.InputDataSet
jar1: /mnt/c/GitHub/cluster/3.0.0/plugins/.installing-3214941709525340169/opensearch-ml-common-3.0.0.0-SNAPSHOT.jar
jar2: /mnt/c/GitHub/cluster/3.0.0/plugins/.installing-3214941709525340169/opensearch-ml-client-3.0.0.0-SNAPSHOT.jar
        at org.opensearch.bootstrap.JarHell.checkClass(JarHell.java:314)
        at org.opensearch.bootstrap.JarHell.checkJarHell(JarHell.java:213)
        at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:667)
        ... 11 more
```
Currently, `ml-client` contains entire `ml-common` and something else.

## What is the expected behavior?
Each exported module should contain unique, non duplicated set of exported classes.

## Do you have any additional context?
Workaround: use `ml-client` shadow jar, which contains all dependencies and produces a security issue. Users of shadow jars are locked in versions of `ml-client` dependencies and change them.

**Comment by ylwu-amzn** [comment]
@Yury-Fridlyand can we close this issue now?

**Comment by Yury-Fridlyand** [comment]
Unfortunately, no. #779 has a workaround, not a fix.
Published shadow jar weights > 250Mb and contains ~all~ **ALL** dependencies inside.

**Comment by vamsi-amazon** [comment]
I am facing this issue when I am trying to include common-utils in sql plugin. For unblocking myself, I am using ml-client instead of common-utils. 
But we cannot release this temp fix in 2.7. Can we get this resolved?

**Comment by ylwu-amzn** [comment]
Fixed in PR #796 , will close this issue.
Feel free to reopen it if you still see jarhell.
